doc/internal/ReleaseInstructions.md

```diff
@@ -95,137 +114,222 @@ Install `gpg-agent` (needed below) e.g. on Ubuntu via: `sudo apt-get install gnu
+  > - If we are doing an RC release, then we will need to create the Vitess Operator RC too. If we are doing a GA release, we're also doing a GA release in the Operator.
+- **Create the Java release.**
+  > - Applies only to GA releases.
+  > - This step is explained in the [Java Packages Deploy & Release](#java-packages-deploy--release) section.
+- **Update the website documentation repository.**
+  > - Applies only to GA and RC releases.
+  > - There are two scripts in the website repository in `./tools/{ga|rc}_release.sh`, use them to update the documentations. The scripts automate:
+  >   - For an RC, we need to create a new version in the sidebar and mark the current version as RC.
+  >   - For a GA, we need to mark the version we are releasing as "Stable" and the next one as "Development".
+- **Publish the blog post on the Vitess website.**
+  > - Applies only to GA releases.
+  > - The corresponding Pull Request was created beforehand during the pre-release. Merge it.
+- **Make sure _arewefastyet_ starts benchmarking the new release.**
+  > - This can be done by visiting [arewefastyet status page](https://benchmark.vitess.io/status).
+  > - New elements should be added to the execution queue.
+  > - After a while, those elements will finish their execution and their status will be green.
+  > - This step is even more important for GA releases as we often include a link to _arewefastyet_ in the blog post.
+  > - The benchmarks need to complete before announcing the blog posts or before they get cross-posted.
+- **Update the release notes on the release branch and on `main`.**
+  > - Two new Pull Requests have to be created.
+  > - One against `main`, it will contain only the new release notes.
+  > - And another against the release branch, this one contains the release notes and the release commit. (The commit on which we did `git tag`) 
+
+### Post-Release
+
+Once the release is over, we need to announce it on both Slack and Twitter. We also want to make sure the blog post was cross-posted, if applicable.
+We need to verify that _arewefastyet_ has finished the benchmark too.
+
+
+### How To Release Vitess
+This section is divided in three parts:
+- How to release an RC: [Pre-Requisites for Release Candidates (`rc`)](#pre-requisites-for-release-candidates-rc).
+- How to release a GA or Patch release: [Pre-Requisites for Releases](#pre-requisites-for-releases).
+- Common to both, how to create the release on the GitHub UI: [#Creating Release or Release Candidate on the GitHub UI](#creating-release-or-release-candidate-on-the-github-ui)
+
+
+#### Pre-Requisites for Release Candidates (`rc`)
 
 > In this example our current version is `v11` and we release the version `v12.0.0-rc1`.
+> Alongside Vitess' release, we also release a new version of the operator.
+> Since we are releasing a release candidate here, the new version of the operator will also be a release candidate.
+> In this example, the new operator version is `2.7.0-rc1`.
```

Comment:
We should use v14 or v15 as the example, since earlier in the doc we say that operator RC releases only apply to Vitess v14+

Reply:
Ideally v15 with the corresponding version number for vitess-operator.

Reply:
Yes, that's a good point.

Reply:
Solved via e5606f0fab